### PR TITLE
Sort annotated classes to process

### PR DESCRIPTION
**Pull request by jprinet**
This is an easy sort which should cover the ordering issue.
There are probably some ways to change the ordering earlier but this approach has the advantage of minimizing the change set.

There is some other randomness on classes iteration but this has no effect on the generated source code.

### CODE REVIEW

**Comment by jprinet** [comment]
@cla-bot check

File: realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ class RealmProxyMediatorGenerator(private val processingEnvironment: ProcessingE
     private val primaryKeyClasses = mutableListOf<QualifiedClassName>()
 
     init {
-        for (metadata in classesToValidate) {
+        for (metadata in classesToValidate.toSortedSet(compareByDescending { it.qualifiedClassName.name })) {
```
Comment (runningcode) [inline realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt:40]: Out of curiousity, what type of `Set` is the set that is passed in here?

Comment (jprinet) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt:40]: A `SortedSet`:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-sorted-set.html

Comment (runningcode) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt:40]: I meant, what is the type of `classesToValidate` which is passed to the constructor of `RealmProxyMediatorGenerator`? I assume it is not already sorted.

Comment (jprinet) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt:40]: makes sense :)
This is just a Set

Comment (runningcode) [reply realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyMediatorGenerator.kt:40]: Thanks, discussed this offline. This looks good!

**Review by runningcode — APPROVED** [review]

**Comment by cmelchior** [comment]
Hi @jprinet Thank you very much for reporting and fixing this. Unfortunately, our CI pipeline doesn't run on external contributions right now, and when I ran tests locally, I noticed that this change broke one of our unit tests. So I took the liberty of making the necessary fixes in this PR https://github.com/realm/realm-java/pull/7616

**Comment by jprinet** [comment]
> Hi @jprinet Thank you very much for reporting and fixing this. Unfortunately, our CI pipeline doesn't run on external contributions right now, and when I ran tests locally, I noticed that this change broke one of our unit tests. So I took the liberty of making the necessary fixes in this PR #7616

Please feel free to add any necessary change. I had some hard times trying to setup a unit test for this change and thought that it would be a better idea to leave it to you anyway.

**Comment by cmelchior** [comment]
A `10.10.0-SNAPSHOT` have been released with this fix: https://github.com/realm/realm-java#using-snapshots

**Comment by jprinet** [comment]
> A `10.10.0-SNAPSHOT` has been released with this fix: https://github.com/realm/realm-java#using-snapshots

awesome, thanks for being so prompt!

**Pull request closed without merging**